test/blackbox-tests/test-cases/public_name-exe.t/run.t

```diff
@@ -48,7 +48,8 @@ With extension, prebuild
   Hello, World!
   $ dune exec -- ./example.exe
   Hello, World!
-  $ sed -i.bak 's/World/Arrakis/' example.ml
```

Comment:
Why is `example.ml` a symlink, anyway? (please don't tell me it's because of sandboxing because that would be terrible)

Reply:
To clarify, if example.ml is a symlink pointing to the build directory then modifying it in a test is just wrong, so "readlink" is also wrong.

Reply:
That is indeed because of sandboxing:

```
 |  $ ls -la example.ml
+|  lrwxr-xr-x 1 rgrinberg staff 88 Jun 15 10:49 example.ml -> ../../../../../../../default/test/blackbox-tests/test-cases/public_name-exe.t/example.ml
```

Reply:
@aalekseyev what should we do about this?

Reply:
@rgrinberg, an unimaginative fix is to simply declare that the test requires sandboxing in the copying mode, but here's a better one:

- remove `example.ml` from git
- add this line at the top of the test:
    `echo 'let () = print_endline "Hello, World!"' > example.ml`

Reply:
@rgrinberg, I pushed that fix. When making it I realized that on linux `sed -i.bak` doesn't actually modify the target of the symlink. See:

```
$ echo hi > a
$ ln -s a b
$ sed -i.bak s/hi/bye/ b
$ cat a
hi
$ ls -al
...
... a
... b
... b.bak -> a
```

So the test was actually well-behaved on Linux already.